### PR TITLE
Sub: make gcs failsafe timeout a parameter

### PR DESCRIPTION
**Pull request by Williangalvani**

### CODE REVIEW

File: ArduSub/defines.h [hunk 1]
```diff
@@ -79,8 +79,8 @@ enum LoggingParameters {
 #ifndef FS_GCS
 # define FS_GCS                        DISABLED
 #endif
-#ifndef FS_GCS_TIMEOUT_MS
-# define FS_GCS_TIMEOUT_MS             2500    // gcs failsafe triggers after this number of milliseconds with no GCS heartbeat
+#ifndef FS_GCS_TIMEOUT_S
+# define FS_GCS_TIMEOUT_S             5.0    // gcs failsafe triggers after this number of milliseconds with no GCS heartbeat
```
Comment (peterbarker) [inline ArduSub/defines.h:83]:
```suggestion
# define FS_GCS_TIMEOUT_S             5.0    // gcs failsafe triggers after this number of seconds with no GCS heartbeat
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: ArduSub/defines.h (hunk 1 above)
Comment (Williangalvani) [reply ArduSub/defines.h]: fixed, thanks!

File: ArduSub/failsafe.cpp [hunk 2]
```diff
@@ -321,7 +321,8 @@ void Sub::failsafe_gcs_check()
     uint32_t tnow = AP_HAL::millis();
 
     // Check if we have gotten a GCS heartbeat recently (GCS sysid must match SYSID_MYGCS parameter)
-    if (tnow - gcs_last_seen_ms < FS_GCS_TIMEOUT_MS) {
+    const float validated_timeout = constrain_float(g.failsafe_gcs_timeout, 2.0, 120);
```
Comment (peterbarker) [inline ArduSub/failsafe.cpp:324]: I really have my reservations about these constraints - particularly the lower one.

This is magic stuff happening in the code, there's nothing the user can know about this from the parameter description.

I don't think it's unreasonable for the user to set outside the bounds you've put in here.

Are you sure you want to bound things here?

Comment (Williangalvani) [reply ArduSub/failsafe.cpp]: I agree with you.
I though I had seen that somewhere else but I cant find it anywhere...
Anyway. code updated, thanks for the review

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**